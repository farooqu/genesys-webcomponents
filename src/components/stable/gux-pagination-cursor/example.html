<h1>gux-pagination-cursor</h1>
<h2>Default (no previous or next page available)</h2>
<div lang="en">
  <div class="table-placeholder"></div>
  <gux-pagination-cursor></gux-pagination-cursor>
</div>

<h2>Next page available</h2>
<div lang="en">
  <div class="table-placeholder"></div>
  <gux-pagination-cursor has-next></gux-pagination-cursor>
</div>

<h2>Next and previous page available</h2>
<div lang="en">
  <div class="table-placeholder"></div>
  <gux-pagination-cursor has-previous has-next></gux-pagination-cursor>
</div>

<h2>Previous page available</h2>
<div lang="en">
  <div class="table-placeholder"></div>
  <gux-pagination-cursor has-previous></gux-pagination-cursor>
</div>

<style>
  .table-placeholder {
    padding: 10px;
    color: #f1f1f1;
    text-align: center;
    background-image: linear-gradient(
      45deg,
      #f1f1f1 25%,
      transparent 25%,
      transparent 50%,
      #f1f1f1 50%,
      #f1f1f1 75%,
      transparent 75%,
      #fff
    );
    background-size: 5px 5px;
  }

  .table-placeholder::before {
    padding: 5px;
    content: 'Table Placeholder';
    background-color: white;
  }
</style>
