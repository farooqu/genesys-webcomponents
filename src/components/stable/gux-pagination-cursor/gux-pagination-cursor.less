@import (reference) '../../../style/color.less';

:host {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-content: stretch;
  align-items: flex-start;
  justify-content: space-between;
}
