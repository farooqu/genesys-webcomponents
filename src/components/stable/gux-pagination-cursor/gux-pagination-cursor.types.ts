export type GuxPaginationCursorDetail = 'previous' | 'next';
