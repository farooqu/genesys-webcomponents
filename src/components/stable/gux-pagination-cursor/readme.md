# gux-pagination-cursor



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description | Type      | Default |
| ------------- | -------------- | ----------- | --------- | ------- |
| `hasNext`     | `has-next`     |             | `boolean` | `false` |
| `hasPrevious` | `has-previous` |             | `boolean` | `false` |


## Events

| Event                       | Description | Type                                |
| --------------------------- | ----------- | ----------------------------------- |
| `guxPaginationCursorchange` |             | `CustomEvent<"next" \| "previous">` |


## Dependencies

### Depends on

- [gux-button-slot-beta](../../beta/gux-button-slot)
- [gux-icon](../gux-icon)

### Graph
```mermaid
graph TD;
  gux-pagination-cursor --> gux-button-slot-beta
  gux-pagination-cursor --> gux-icon
  style gux-pagination-cursor fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
