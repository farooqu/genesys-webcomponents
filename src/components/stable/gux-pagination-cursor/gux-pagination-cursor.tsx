import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  JSX,
  Prop
} from '@stencil/core';

import { trackComponent } from '../../../usage-tracking';
import { buildI18nForComponent, GetI18nValue } from '../../../i18n';

import translationResources from './i18n/en.json';
import { GuxPaginationCursorDetail } from './gux-pagination-cursor.types';

@Component({
  styleUrl: 'gux-pagination-cursor.less',
  tag: 'gux-pagination-cursor',
  shadow: true
})
export class GuxPaginationCursor {
  private i18n: GetI18nValue;

  @Element()
  private root: HTMLElement;

  @Prop()
  hasPrevious: boolean = false;

  @Prop()
  hasNext: boolean = false;

  @Event()
  private guxPaginationCursorchange: EventEmitter<GuxPaginationCursorDetail>;

  private onButtonClick(paginationDetail: GuxPaginationCursorDetail): void {
    if (
      (paginationDetail === 'previous' && this.hasPrevious) ||
      (paginationDetail === 'next' && this.hasNext)
    ) {
      this.guxPaginationCursorchange.emit(paginationDetail);
    }
  }

  async componentWillLoad(): Promise<void> {
    trackComponent(this.root);

    this.i18n = await buildI18nForComponent(this.root, translationResources);
  }

  render(): JSX.Element {
    return [
      <gux-button-slot-beta accent="secondary">
        <button
          type="button"
          title={this.i18n('previous')}
          disabled={!this.hasPrevious}
          onClick={() => this.onButtonClick('previous')}
        >
          <gux-icon decorative iconName="ic-chevron-small-left"></gux-icon>
        </button>
      </gux-button-slot-beta>,
      <gux-button-slot-beta accent="secondary">
        <button
          type="button"
          title={this.i18n('next')}
          disabled={!this.hasNext}
          onClick={() => this.onButtonClick('next')}
        >
          <gux-icon decorative iconName="ic-chevron-small-right"></gux-icon>
        </button>
      </gux-button-slot-beta>
    ] as JSX.Element;
  }
}
