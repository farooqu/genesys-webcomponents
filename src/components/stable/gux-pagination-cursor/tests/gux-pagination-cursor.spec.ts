import { newSpecPage } from '@stencil/core/testing';
import { GuxPaginationCursor } from '../gux-pagination-cursor';

const components = [GuxPaginationCursor];
const language = 'en';

describe('gux-pagination-cursor', () => {
  describe('#render', () => {
    [
      '<gux-pagination-cursor></gux-pagination-cursor>',
      '<gux-pagination-cursor has-next></gux-pagination-cursor>',
      '<gux-pagination-cursor has-previous has-next></gux-pagination-cursor>',
      '<gux-pagination-cursor has-previous></gux-pagination-cursor>'
    ].forEach((html, index) => {
      it(`should render as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.rootInstance).toBeInstanceOf(GuxPaginationCursor);
        expect(page.root).toMatchSnapshot();
      });
    });
  });

  describe('guxPaginationCursorchange', () => {
    it('should fire guxPaginationCursorchange(previous) event when enabled previous button is clicked', async () => {
      const html =
        '<gux-pagination-cursor has-previous></gux-pagination-cursor>';
      const page = await newSpecPage({ components, html, language });
      const element = page.root as HTMLElement;
      const [previousButton] = Array.from(
        element.shadowRoot.querySelectorAll('button')
      ) as HTMLElement[];
      const guxPaginationCursorchangeSpy = jest.fn();

      page.win.addEventListener(
        'guxPaginationCursorchange',
        guxPaginationCursorchangeSpy
      );

      previousButton.click();
      await page.waitForChanges();

      expect(guxPaginationCursorchangeSpy).toHaveBeenCalledWith(
        expect.objectContaining({ detail: 'previous' })
      );
    });

    it('should fire guxPaginationCursorchange(next) event when enabled next button is clicked', async () => {
      const html = '<gux-pagination-cursor has-next></gux-pagination-cursor>';
      const page = await newSpecPage({ components, html, language });
      const element = page.root as HTMLElement;
      const [, nextButton] = Array.from(
        element.shadowRoot.querySelectorAll('button')
      ) as HTMLElement[];
      const guxPaginationCursorchangeSpy = jest.fn();

      page.win.addEventListener(
        'guxPaginationCursorchange',
        guxPaginationCursorchangeSpy
      );

      nextButton.click();
      await page.waitForChanges();

      expect(guxPaginationCursorchangeSpy).toHaveBeenCalledWith(
        expect.objectContaining({ detail: 'next' })
      );
    });

    it('should not fire guxPaginationCursorchange(previous) event when disabled previous button is clicked', async () => {
      const html = '<gux-pagination-cursor></gux-pagination-cursor>';
      const page = await newSpecPage({ components, html, language });
      const element = page.root as HTMLElement;
      const [previousButton] = Array.from(
        element.shadowRoot.querySelectorAll('button')
      ) as HTMLElement[];
      const guxPaginationCursorchangeSpy = jest.fn();

      page.win.addEventListener(
        'guxPaginationCursorchange',
        guxPaginationCursorchangeSpy
      );

      previousButton.click();
      await page.waitForChanges();

      expect(guxPaginationCursorchangeSpy).not.toHaveBeenCalled();
    });

    it('should not fire guxPaginationCursorchange(next) event when disabled next button is clicked', async () => {
      const html = '<gux-pagination-cursor></gux-pagination-cursor>';
      const page = await newSpecPage({ components, html, language });
      const element = page.root as HTMLElement;
      const [, nextButton] = Array.from(
        element.shadowRoot.querySelectorAll('button')
      ) as HTMLElement[];
      const guxPaginationCursorchangeSpy = jest.fn();

      page.win.addEventListener(
        'guxPaginationCursorchange',
        guxPaginationCursorchangeSpy
      );

      nextButton.click();
      await page.waitForChanges();

      expect(guxPaginationCursorchangeSpy).not.toHaveBeenCalled();
    });
  });
});
