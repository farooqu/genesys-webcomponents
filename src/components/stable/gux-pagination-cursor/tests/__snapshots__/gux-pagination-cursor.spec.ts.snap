// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-pagination-cursor #render should render as expected (1) 1`] = `
<gux-pagination-cursor>
  <mock:shadow-root>
    <gux-button-slot-beta accent="secondary">
      <button disabled="" title="Previous" type="button">
        <gux-icon decorative="" iconname="ic-chevron-small-left"></gux-icon>
      </button>
    </gux-button-slot-beta>
    <gux-button-slot-beta accent="secondary">
      <button disabled="" title="Next" type="button">
        <gux-icon decorative="" iconname="ic-chevron-small-right"></gux-icon>
      </button>
    </gux-button-slot-beta>
  </mock:shadow-root>
</gux-pagination-cursor>
`;

exports[`gux-pagination-cursor #render should render as expected (2) 1`] = `
<gux-pagination-cursor has-next="">
  <mock:shadow-root>
    <gux-button-slot-beta accent="secondary">
      <button disabled="" title="Previous" type="button">
        <gux-icon decorative="" iconname="ic-chevron-small-left"></gux-icon>
      </button>
    </gux-button-slot-beta>
    <gux-button-slot-beta accent="secondary">
      <button title="Next" type="button">
        <gux-icon decorative="" iconname="ic-chevron-small-right"></gux-icon>
      </button>
    </gux-button-slot-beta>
  </mock:shadow-root>
</gux-pagination-cursor>
`;

exports[`gux-pagination-cursor #render should render as expected (3) 1`] = `
<gux-pagination-cursor has-next="" has-previous="">
  <mock:shadow-root>
    <gux-button-slot-beta accent="secondary">
      <button title="Previous" type="button">
        <gux-icon decorative="" iconname="ic-chevron-small-left"></gux-icon>
      </button>
    </gux-button-slot-beta>
    <gux-button-slot-beta accent="secondary">
      <button title="Next" type="button">
        <gux-icon decorative="" iconname="ic-chevron-small-right"></gux-icon>
      </button>
    </gux-button-slot-beta>
  </mock:shadow-root>
</gux-pagination-cursor>
`;

exports[`gux-pagination-cursor #render should render as expected (4) 1`] = `
<gux-pagination-cursor has-previous="">
  <mock:shadow-root>
    <gux-button-slot-beta accent="secondary">
      <button title="Previous" type="button">
        <gux-icon decorative="" iconname="ic-chevron-small-left"></gux-icon>
      </button>
    </gux-button-slot-beta>
    <gux-button-slot-beta accent="secondary">
      <button disabled="" title="Next" type="button">
        <gux-icon decorative="" iconname="ic-chevron-small-right"></gux-icon>
      </button>
    </gux-button-slot-beta>
  </mock:shadow-root>
</gux-pagination-cursor>
`;
